thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/auth/ThirdEyeAuthFilter.java

```diff
@@ -109,7 +108,7 @@ public void filter(ContainerRequestContext requestContext) {
     }
   }
 
-  private boolean isAuthenticated(ContainerRequestContext containerRequestContext, ThirdEyePrincipal principal) {
+  private ThirdEyePrincipal getPrincipal(ContainerRequestContext containerRequestContext) {
```

Comment:
`Optional<ThirdEyePrincipal>` ?

Reply:
I know. I tried that initially. However, the way things are written, I would have had to do a bunch of `optional.get()` which defeats the purpose. The other option is to rewrite the logic to make it more optional friendly, but didn't want to do that in this PR.

Reply:
understood.